packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -113,6 +114,39 @@ class Config:
     """User inputs expected by this provider when initializing it. Each `Field` `f`
     should be passed in the constructor as a keyword argument, keyed by `f.key`."""
 
+    prompt_templates: ClassVar[Dict[str, PromptTemplate]] = {
```

Comment:
This should not be a class attribute, but instead an instance attribute. Class attributes are shared across all instances of `BaseProvider`. This means that the following unintended behavior can occur:

```
# ai21_provider and openai_provider are two separate instances of BaseProvider,
# such that ai21_provider.prompt_template("html") := A
openai_provider.update_prompt_template("html", B)

# this now evaluates to B instead of A
ai21_provider.prompt_template("html")
```

I had called this out in a previous review comment.

Instead, move the definition to the constructor:

```
self.prompt_templates = {
  # replace this with the dictionary
  ...
}
```

At the class' top level (where I have highlighted), the `prompt_templates` attribute should still be _declared but not defined_:

```
prompt_templates: Dict[str, PromptTemplate]
```